viz_scripts/scaffolding.py

```diff
@@ -191,28 +207,48 @@ def translate_labels(labels):
                 translation = translations.get(value)
                 translation_mapping[value] = translation
             return defaultdict(lambda: 'Other', translation_mapping)
-        dic_mapping = translate_labels(dynamic_labels[label_type])
+        dic_mapping = translate_labels(dynamic_labels.get(label_type, ''))
         return dic_mapping
 
 # Function: Maps "MODE", "PURPOSE", and "REPLACED_MODE" to colors.
 # Input: dynamic_labels, dic_re, and dic_pur
 # Output: Dictionary mapping between color with mode/purpose/sensed
-def mapping_color_labels(dynamic_labels, dic_re, dic_pur):
-    sensed_values = ["WALKING", "BICYCLING", "IN_VEHICLE", "AIR_OR_HSR", "UNKNOWN", "OTHER", "Other"]
-    if len(dynamic_labels) > 0:
-        mode_values = list(mapping_labels(dynamic_labels, "MODE").values()) if "MODE" in dynamic_labels else []
-        replaced_mode_values = list(mapping_labels(dynamic_labels, "REPLACED_MODE").values()) if "REPLACED_MODE" in dynamic_labels else []
-        purpose_values = list(mapping_labels(dynamic_labels, "PURPOSE").values()) + ['Other'] if "PURPOSE" in dynamic_labels else []
-        combined_mode_values = mode_values + replaced_mode_values + ['Other']
-    else:
-        combined_mode_values = (list(OrderedDict.fromkeys(dic_re.values())) + ['Other'])
-        purpose_values = list(OrderedDict.fromkeys(dic_pur.values()))
+async def mapping_color_labels(dynamic_labels, language="en"):
+    # Load default options from e-mission-common
+    labels = await read_json_resource("label-options.default.json")
+    sensed_values = ["WALKING", "BICYCLING", "IN_VEHICLE", "AIR_OR_HSR", "UNKNOWN", "OTHER", "other"]
 
-    colors_mode = dict(zip(combined_mode_values, plt.cm.tab20.colors[:len(combined_mode_values)]))
+    # If dynamic_labels are provided, then we will use the dynamic labels for mapping
+    if len(dynamic_labels) > 0:
+        labels = dynamic_labels
+
+    # Load base mode values and purpose values
+    mode_values =  [mode["value"] for mode in labels["MODE"]] if "MODE" in labels else []
+    purpose_values = [mode["value"] for mode in labels["PURPOSE"]] if "PURPOSE" in labels else []
+    replaced_values = [mode["value"] for mode in labels["REPLACED_MODE"]] if "REPLACED_MODE" in labels else []
+
+    # Mapping between mode values and base_mode OR baseMode (backwards compatibility)
```

Comment:
I do think it would be best to split this off from the colors, I see that they both rely on the `label-options.default.json` but since they are otherwise separate tasks I think that separate functions are warranted

Reply:
Handled this in a new function `translate_values_to_labels()`

Reply:
I'm confused; I see that there is a new function, but it is not invoked here to do the mapping. I see the function invoked in the notebooks and the results passed in to the plots, but not here in scaffolding. If this is no longer relevant, the comment should be removed.